src/blueapi/client/numtracker.py

```diff
@@ -0,0 +1,107 @@
+import logging
+from collections.abc import Mapping
+from pathlib import Path
+from textwrap import dedent
+
+import requests
+from pydantic import Field
+
+from blueapi.utils import BlueapiBaseModel
+
+
+class DirectoryPath(BlueapiBaseModel):
+    """
+    Directory location and associated metadata
+    """
+
+    instrument: str
+    instrument_session: str = Field(alias="instrumentSession")
+    path: Path
+
+
+class ScanPaths(BlueapiBaseModel):
+    """
+    Description of where to write scan data
+    """
+
+    scan_file: str = Field(alias="scanFile")
+    scan_number: int = Field(alias="scanNumber")
+    directory: DirectoryPath
+
+
+class NumtrackerScanMutationResponse(BlueapiBaseModel):
+    """
+    Response from numtracker server when creating a new scan
+    """
+
+    scan: ScanPaths
+
+
+class NumtrackerClient:
+    """
+    Client for numtracker: https://github.com/DiamondLightSource/numtracker
+    Can be used to assign new scan numbers.
+    """
+
+    def __init__(
+        self,
+        url: str,
+        headers: Mapping[str, str],
```

Comment:
```suggestion
```
Looks like these headers and are overriden by calling set_headers later?
Could remove them from the constructor and make creating the Client slightly easier.

Reply:
Should still do `self._headers = {}` in the constructor to prevent `AttributeError` being raised if `set_headers` was never called

Reply:
Done